src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -408,6 +448,34 @@ by tidy
   coprod.map h k ≫ coprod.desc f g = coprod.desc (h ≫ f) (k ≫ g) :=
 by tidy
 
+@[simp, reassoc]
+lemma coprod.inl_codiag {X : C} : coprod.inl ≫ codiag X = 𝟙 X :=
+by simp
+
+@[simp, reassoc]
+lemma coprod.inr_codiag {X : C} : coprod.inr ≫ codiag X = 𝟙 X :=
+by simp
```

Comment:
Are these necessary, if they're already provable by `simp`? If so, perhaps a quick comment explaining why?

Reply:
I think we can do without it. Same for `prod.diag_fst` and `prod.diag_snd`.